src/query/functions/src/scalars/map.rs

```diff
@@ -229,6 +241,125 @@ pub fn register(registry: &mut FunctionRegistry) {
         |map, _| map.len() as u64,
     );
 
+    registry.register_function_factory("map_delete", |_, args_type| {
+        if args_type.len() < 2 {
+            return None;
+        }
+
+        let (map_key_type, _map_value_type) = match args_type[0].remove_nullable() {
+            DataType::Map(box DataType::Tuple(type_tuple)) => {
+                if type_tuple.len() == 2 {
+                    (type_tuple[0].clone(), type_tuple[1].clone())
+                } else {
+                    return None;
+                }
+            }
+            DataType::EmptyMap => (DataType::EmptyMap, DataType::EmptyMap),
+            _ => {
+                return None;
+            }
+        };
+
+        let mut in_key_types: Option<DataType> = None;
+
+        for arg_type in args_type.iter().skip(1) {
+            match arg_type {
+                DataType::Array(box key_type) => {
```

Comment:
We follow the snowflake syntax, the argument types should be same as the key type, not an array type.
Please refer to this document:
https://docs.snowflake.com/en/sql-reference/functions/map_delete#syntax

Reply:
Thankyou. Yes, I realised it latter.

Reply:
you can write like this, `map_key_type` is `None` if the first argument is `EmptyMapType`.
```rust
if let Some(key_type) = map_key_type {
    for arg_type in args_type.iter().skip(1) {
        if arg_type != &key_type {
            return None;
        }
    }
} else {
    let key_type = args_type[1];
    if !key_type.is_boolean()
        && !key_type.is_string()
        && !key_type.is_numeric()
        && !key_type.is_decimal()
        && !key_type.is_date_or_date_time() {
    }
    for arg_type in args_type.iter().skip(2) {
        if arg_type != &key_type {
            return None;
        }
    }
}
```